server/protocols/protocols-smtp/src/main/java/org/apache/james/smtpserver/fastfail/URIScanner.java

```diff
@@ -84,7 +84,7 @@
 
     private static final String ctrl = "\\000-\\037";
 
-    private static final String cr_list = "\\n\\015";
+    private static final String cr_list = "\\n\\r";
```

Comment:
that's not '\r', that's '\\\\r', so it is wanted?

Reply:
We may check that a test covers that.

Reply:
So there is no test for this \r thing. URIScanner is a big thing and i'm not ready to handle coverage on that.
Could you elaborate on the reason you thing it's not ok ? unicode escape is \\ + a number (here an octal one). 015 being \r, I don't see where the problem is.
So either we agree to let it that way, either i'll disable the rule about unicode escape for now and revert this change. WDYT ?

Reply:
as discussed, \\\\r is different from \\\\015, because the \ is escaped.
But in this case it's the same because it's used only in a regexp.